packages/core/src/networks.ts

```diff
@@ -140,6 +142,12 @@ export const calculateMerkleLeafGas = (chainId: bigint, foundryGas: string) => {
   const network = SPHINX_NETWORKS.find((n) => n.chainId === chainId)
 
   if (network?.hardcodedMerkleLeafGas) {
+    if (BigInt(foundryGas) > BigInt(network?.hardcodedMerkleLeafGas)) {
+      throw new InvariantError(
+        'Transaction too large to be executed on Pangolin'
```

Comment:
The error message says Pangolin but it can be triggered on other networks, e.g. Moonbeam